Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6970,7 +6970,8 @@ void WebPageProxy::mouseDidMoveOverElement(WebHitTestResultData&& hitTestResultD
 {
     m_lastMouseMoveHitTestResult = API::HitTestResult::create(hitTestResultData);
     m_uiClient->mouseDidMoveOverElement(*this, hitTestResultData, modifiers, m_process->transformHandlesToObjects(userData.object()).get());
-    setToolTip(hitTestResultData.toolTipText);
+    if (m_pageClient)
```

Comment:
pageClient is not meant to be null checked, which is why pageClient() returns a reference. It would be better to make sure the page is not closed (do a `WebPageProxy::isClosed()` check).

Reply:
Also, we may want to do this check earlier. Probably not much point in doing the earlier work if the page is closed.

Reply:
> pageClient is not meant to be null checked, which is why pageClient() returns a reference

Yeah it did seem sort of an anti-pattern to do so inside `WebPageProxy`

Reply:
> It would be better to make sure the page is not closed (do a WebPageProxy::isClosed() check).

The reason we hit this issue is because we're async dispatching a completion handler that calls `mouseDidMoveOverElement`, so we can't consult `WebPageProxy::isClosed()` when that work is scheduled.

Thoughts?

Reply:
Hmmm, maybe the completion handler itself could consult `WebPageProxy::isClosed()` before calling into `WebPageProxy::mouseDidMoveOverElement()`

Reply:
I think we can consult either `isClosed` or `hasRunningProcess` (which is slightly more conservative) in the completion handler.

Reply:
Either early return in this function or before calling this function in the lambda. I don’t feel strongly either way. However, it should be a IsClosed() check.

Reply:
Thanks for the catch again, fixed in https://github.com/WebKit/WebKit/commit/3547821b6b8c0933194eb6bb9ec328c6bfd166ae